dimension.go

```diff
@@ -0,0 +1,381 @@
+package tiledb
+
+/*
+#cgo LDFLAGS: -ltiledb
+#include <tiledb/tiledb.h>
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"strconv"
+	"unsafe"
+)
+
+// Dimension is tiledb dimension
+type Dimension struct {
+	tiledbDimension *C.tiledb_dimension_t
+	context         *Context
+}
+
+// NewDimension alloc a new dimension
+func NewDimension(context *Context, name string, domain interface{}, extent interface{}) (*Dimension, error) {
+	dimension := Dimension{context: context}
+	var cname *C.char = C.CString(name)
+	defer C.free(unsafe.Pointer(cname))
+
+	if reflect.TypeOf(domain).Kind() != reflect.Slice {
```

Comment:
Is there a way here to check if the domain slice type == extent slice type?  Probably best to enforce this as issues with auto type conversion can be tricky to debug.

Reply:
A safety check for this is simple to add, I'll do that next.